server/src/main/java/org/opensearch/index/engine/NRTReplicationEngine.java

```diff
@@ -119,6 +119,23 @@ public void onAfterTranslogSync() {
         }
     }
 
+    private NRTReplicationReaderManager buildReaderManager() throws IOException {
+        return new NRTReplicationReaderManager(
+            OpenSearchDirectoryReader.wrap(getDirectoryReader(), shardId),
+            store::incRefFileDeleter,
+            (files) -> {
+                store.decrefFileDeleter(files);
+                try {
+                    store.cleanupAndPreserveLatestCommitPoint("On reader closed", getLatestSegmentInfos());
```

Comment:
Forgot to call this out in my original pr. I'm worried about this cleanup method being too expensive. It will be invoked with every reader deletion and internally it does a disk read of the latest commit. Maybe worth a micro benchmark with iops? Alternatively we can refactor the cleanup method in this instance to use a reference to latest commit instead of reading from disk. Or, we update deleter to delete specifically the unreferenced files on decref, and incref as new segments arrive and commits are made.

Reply:
I see this failure now without the [optimization](https://github.com/opensearch-project/OpenSearch/pull/6765/commits/24ccdab5eecc99734eb8863291b825edde23e043), seems like something changed in main which is not working well

Reply:
@nknize : Do you see this as a big problem? Can we take this up as a follow up post segrep GA release (2.7) ?

Reply:
Since this is not impacting the functionality, I think can add it as fast follow if needed

Reply:
Added couple of changes in https://github.com/opensearch-project/OpenSearch/pull/7130 that avoids unnecessary reads during replica commit. 

Remaining cleanup and optimization will take up in https://github.com/opensearch-project/OpenSearch/pull/7044